contracts/finance/Vesting_2.sol

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {MathUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {Counters} from "@openzeppelin/contracts/utils/Counters.sol";
+
+import {PRECISION} from "../utils/Globals.sol";
+
+abstract contract VestingWallet is Initializable {
+    using MathUpgradeable for uint256;
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.UintSet;
+    using Counters for Counters.Counter;
+
+    struct Schedule {
+        uint256 startTime;
```

Comment:
If you are going this road then dynamic variables have to be moved to the `Vesting` struct. Also calculation formula should become a part of the schedule instead